clients/admin-ui/src/features/config-wizard/ScanResultsForm.tsx

```diff
@@ -33,26 +36,38 @@ const ALL_COLUMNS: ColumnMetadata[] = [
 const ScanResultsForm = () => {
   const systems = useAppSelector(selectSystemsForReview);
   const dispatch = useAppDispatch();
+  const router = useRouter();
   const {
     isOpen: isWarningOpen,
     onOpen: onWarningOpen,
     onClose: onWarningClose,
   } = useDisclosure();
+  const [upsertSystems] = useUpsertSystemsMutation();
   const [selectedSystems, setSelectedSystems] = useState<System[]>(systems);
+  const features = useFeatures();
   const [selectedColumns, setSelectedColumns] =
     useState<ColumnMetadata[]>(ALL_COLUMNS);
 
+  const createSystems = async () => {
+    await upsertSystems(selectedSystems);
+  };
+
   const confirmRegisterSelectedSystems = () => {
     dispatch(chooseSystemsForReview(selectedSystems.map((s) => s.fides_key)));
-    dispatch(changeStep());
+    createSystems();
+
+    if (features.plus) {
+      router.push(`/datamap`);
+    }
+
+    router.push(`/system`);
   };
 
   const handleSubmit = () => {
     if (systems.length > selectedSystems.length) {
       onWarningOpen();
-    } else {
-      confirmRegisterSelectedSystems();
     }
+    confirmRegisterSelectedSystems();
```

Comment:
hmm are you sure we don't need the `else` here? we wouldn't want to open the warning AND confirm selected systems. not sure if that's what happens, or if opening the warning somehow breaks us out of the function?

Reply:
I just tested it out, and confirmed removing the `else` does make clicking "register" just register the systems without first waiting for the warning to be accepted